### PR TITLE
Handle updateMask for Update operations

### LINKED ISSUE #300: Updating a Secret throws: "Error 400: Field [update_mask] is required"
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

Updating a Secret resource with a new label throws the error:
```
google-native:secretmanager/v1:Secret (my-secret):
    error: error sending request: googleapi: Error 400: Field [update_mask] is required.: "https://secretmanager.googleapis.com/v1/projects/pulumi-ce-team/secrets/my-secret" map[labels:map[label:my-label] replication:map[automatic:map[]]]
```

This appears to be a variation of this issue: https://github.com/pulumi/pulumi-google-native/issues/95 

### Steps to reproduce

1. Run `pulumi up` on  code below with labels property commented out.
2. Uncomment labels property
3. Run `pulumi up` again and see that the resource will be updated and allow update to proceed
4. Receive error.

Code:
```
from pulumi_google_native.secretmanager import v1 as gcp

secret = gcp.Secret("my-secret",
                    secret_id="my-secret",
                    replication=gcp.ReplicationArgs(automatic=gcp.AutomaticArgs()),
                   ### UNCOMMENT ### labels={"label":"my-label"}
                )
```

Expected: 
The update should proceed as expected
Actual: 
Failed due to `update_mask` error.

### PR DESCRIPTION
**Pull request by viveklak**
Fixes #300

### CODE REVIEW

File: provider/pkg/gen/schema.go [hunk 1]
```diff
@@ -915,7 +916,10 @@ func (g *packageGenerator) genProperties(typeName string, typeSchema *discovery.
 		}
 
 		apiProp := resources.CloudAPIProperty{
-			Ref:                  typeSpec.Ref,
+			Ref:      typeSpec.Ref,
+			Format:   prop.Format,
+			Required: prop.Required || isRequired(prop.Description),
```
Comment (lblackstone) [inline provider/pkg/gen/schema.go:921]: Should we refactor `isRequired` to take the property and internally handle both `prop.Required` and `prop.Description`?

File: provider/pkg/provider/provider.go [hunk 2]
```diff
@@ -863,3 +901,12 @@ func parseCheckpointObject(obj resource.PropertyMap) resource.PropertyMap {
 
 	return nil
 }
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+func ToSnakeCase(str string) string {
```
Comment (lblackstone) [inline provider/pkg/provider/provider.go:908]: Should we use the `strcase` module for this instead?

**Review by lblackstone — APPROVED** [review]
🎉

**Comment by ColeSiegelTR** [comment]
Thanks for the fix! What version can I find this in? Is it in `0.17.1`?

**Comment by lblackstone** [comment]
> What version can I find this in? Is it in `0.17.1`?

Yep, fixed in `v0.17.1`.

**Comment by viveklak** [comment]
@ColeSiegelTR - I reopened #254. Will update there.

**Pull request merged**